lib/presentation/shared_widgets/phone_input.dart

```diff
@@ -0,0 +1,190 @@
+/// Phone input with country search and selection
+class PhoneInput extends StatefulWidget {
+  /// Callback for realtime input validation
+  final Function(bool)? isValid;
+
+  /// Callback with generated phone number
+  final Function(PhoneResponse)? onChange;
+
+  final PhoneResponse? phone;
+
+  const PhoneInput({Key? key, this.isValid, this.onChange, this.phone})
+      : super(key: key);
+
+  @override
+  _PhoneInputState createState() => _PhoneInputState();
+}
+
+class _PhoneInputState extends State<PhoneInput> {
+  Country? _selected;
+
+  final plugin = PhoneNumberUtil();
+  final _phoneNumberController = TextEditingController();
+  bool validatedNumber = false;
+
+  @override
+  void initState() {
+    if (widget.phone != null) {
+      final country =
+          countries.where((country) => country.code == widget.phone?.country);
+
+      /// Strip country
+      if (country.isNotEmpty) {
+        _selected = country.first;
+      } else {
+        _selected = countries.where((country) => country.code == "NL").first;
+      }
+
+      /// Strip phone
+      final contact = widget.phone?.contact.split(" ");
+
+      if (contact?.isNotEmpty == true && contact?.length == 2) {
+        _phoneNumberController.text = contact?.last ?? "";
+      }
+    } else {
+      _selected = countries.where((country) => country.code == "NL").first;
+    }
+
+    _phoneNumberController.addListener(() {
+      if (_phoneNumberController.value.text.isNotEmpty) {
+        setState(() {
+          _validatePhone(_selected!, _phoneNumberController.value.text);
+        });
+      }
+    });
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        IntrinsicHeight(
+          child: Row(
+            crossAxisAlignment: CrossAxisAlignment.stretch,
+            children: [
+              InkWell(
+                onTap: _showCountrySelectDialog,
+                child: Container(
+                  padding: const EdgeInsets.symmetric(horizontal: 10),
+                  color: Theme.of(context).inputDecorationTheme.fillColor,
+                  child: Row(
+                    children: [
+                      Image.asset(
+                        'icons/flags/png/${_selected?.code.toLowerCase()}.png',
+                        package: 'country_icons',
+                        width: 24.0,
+                      ),
+                      const SizedBox(width: 10),
+                      Text(
+                        _selected?.dial_code ?? "",
+                        style: const TextStyle(fontSize: 16.0),
+                        key:const Key("dial_code"),
+                      ),
+                      const Icon(Icons.expand_more_outlined),
+                    ],
+                  ),
+                ),
+              ),
+              const SizedBox(width: 10),
+              Expanded(
+                child: TextFormField(
+                  controller: _phoneNumberController,
```

Comment:
Can we format a phone number with the correct format (region-based?) with a listener on this controller or in the onChange method? Do you want to look into this or should I? :D

Reply:
I had actually used the `PhoneResponse` class in `onChange` callback that provides the county code and formatted phone number (with region) whenever anything changes. Is that what you meant ??